libs/confman/inc/public/tfc/confman/file_storage.hpp

```diff
@@ -12,6 +15,9 @@
 
 namespace tfc::confman {
 
+auto write_to_file(const std::filesystem::path& file_path, std::string_view file_contents) -> std::error_code;
+auto write_and_delete_old_files(std::string_view file_content, std::filesystem::path const& file_path) -> std::error_code;
```

Comment:
I really don't like the name of this function. Does it need to exist?
Can the users not just 
```
write_to_file();
apply_retention_policy():
```
The delete old files is really a lie. These are not necessarily old files.

Reply:
I agree that the name should rather be `apply_retention_policy` but I disagree that the function should not be used. If this function is skipped then the header file needs to be exposed to more functions instead of just these two, in my opinion it is cleaner to only expose these 2.